test/v1/controllers/ListEmploymentsControllerSpec.scala

```diff
@@ -0,0 +1,228 @@
+      rel = SELF
+    )
+
+  val addCustomEmploymentLink: Link =
+    Link(
+      href = s"/individuals/income-received/employments/$nino/$taxYear",
+      method = POST,
+      rel = ADD_CUSTOM_EMPLOYMENT
+    )
+
+  val listEmploymentsLink: Link =
+    Link(
+      href = s"/individuals/income-received/employments/$nino/$taxYear",
+      method = GET,
+      rel = SELF
+    )
+
+  private val hmrcEmploymentModel = Employment(
+    employmentId = "4557ecb5-fd32-48cc-81f5-e6acd1099f3c",
+    employerName = "Vera Lynn",
+    dateIgnored = Some("2020-06-17T10:53:38Z")
+  )
+
+  private val customEmploymentModel = Employment(
+    employmentId = "4557ecb5-fd32-48cc-81f5-e6acd1099f3c",
+    employerName = "Vera Lynn"
+  )
+
+  private val listEmploymentsResponseModel = ListEmploymentResponse(
+    Some(Seq(hmrcEmploymentModel,hmrcEmploymentModel)),
+    Some(Seq(customEmploymentModel,customEmploymentModel))
+  )
+
+  private val hateoasResponse = ListEmploymentResponse(
+    Some(Seq(
+      HateoasWrapper(
+        hmrcEmploymentModel,
+        Seq(retrieveEmploymentLink)
+      ),
+      HateoasWrapper(
+        hmrcEmploymentModel,
+        Seq(retrieveEmploymentLink)
+      )
+    )),
+    Some(Seq(
+      HateoasWrapper(
+        customEmploymentModel,
+        Seq(retrieveEmploymentLink)
+      ),
+      HateoasWrapper(
+        customEmploymentModel,
+        Seq(retrieveEmploymentLink)
+      )
+    ))
+  )
+
+  private val mtdResponse = mtdResponseWithCustomHateoas(nino, taxYear, employmentId) ++ baseHateoas(nino, taxYear)
+
+  trait Test {
+    val hc = HeaderCarrier()
+
+    val controller = new ListEmploymentsController(
+      authService = mockEnrolmentsAuthService,
+      lookupService = mockMtdIdLookupService,
+      requestParser = mockListEmploymentsRequestParser,
+      service = mockListEmploymentsService,
+      hateoasFactory = mockHateoasFactory,
+      cc = cc
+    )
+
+    MockedMtdIdLookupService.lookup(nino).returns(Future.successful(Right("test-mtd-id")))
+    MockedEnrolmentsAuthService.authoriseUser()
+  }
+
+  "ListEmploymentsController" should {
+    "return OK" when {
+      "happy path" in new Test {
+
+        MockListEmploymentsRequestParser
+          .parse(rawData)
+          .returns(Right(requestData))
+
+        MockListEmploymentsService
+          .listEmployments(requestData)
+          .returns(Future.successful(Right(ResponseWrapper(correlationId, listEmploymentsResponseModel))))
+
+        MockHateoasFactory
+          .wrapList(listEmploymentsResponseModel, ListEmploymentHateoasData(nino, taxYear))
+          .returns(HateoasWrapper(hateoasResponse,
+            Seq(
+              addCustomEmploymentLink,
+              listEmploymentsLink
+            )
+          ))
+
+        val result: Future[Result] = controller.listEmployments(nino, taxYear)(fakeGetRequest)
+
+        status(result) shouldBe OK
+        contentAsJson(result) shouldBe mtdResponse.as[JsObject] ++ mtdResponse.as[JsObject]
```

Comment:
Why are you adding the expected response to itself? I'm not entirely sure what this even does since removing it doesn't cause the test to fail.
```suggestion
        contentAsJson(result) shouldBe mtdResponse.as[JsObject]
```

Reply:
Made changes to this so don't know why it's still there. Will revert it back.

Reply:
Changed to mtdResponse instead.